src/raglite/_database.py

```diff
@@ -129,10 +137,55 @@ def __repr__(self) -> str:
             indent=4,
         )
 
-    def __str__(self) -> str:
-        """Context representation of this chunk."""
+    @property
+    def content(self) -> str:
+        """Return this chunk's contextual heading and body."""
         return f"{self.headings.strip()}\n\n{self.body.strip()}".strip()
 
+    def __str__(self) -> str:
+        """Return this chunk's content."""
+        return self.content
+
+
+@dataclass
+class ChunkSpan:
+    """A consecutive sequence of chunks from a single document."""
+
+    chunks: list[Chunk]
+    document: Document
+
+    def to_xml(self, index: int | None = None) -> str:
+        """Convert this chunk span to an XML representation.
+
+        The XML representation follows Anthropic's best practices [1].
+
+        [1] https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/long-context-tips
+        """
+        if not self.chunks:
+            return ""
+        index_attribute = f' index="{index}"' if index is not None else ""
+        xml = "\n".join(
+            [
+                f'<document{index_attribute} id="{self.document.id}" from_chunk_id="{self.chunks[0].id}" to_chunk_id="{self.chunks[-1].id}">',
```

Comment:
I like it, just not sure if you can retrieve the full list of chunks like that, as the chunk ids are not consecutive in general (or are they?)

Reply:
It is assumed that when you create a `ChunkSpan`, you pass in the (contiguous) chunks in the correct order.

In this case, it is `retrieve_chunk_spans` that is responsible for ordering the chunks within each `ChunkSpan` correctly.

The `ChunkSpan`s themselves are ordered according to the aggregate ranking score of the underlying `Chunk`s.

Reply:
I meant this `self.chunks[0].id`, AFAIK it is a random string (uuid or similar), am I wrong?, therefore we can't know the in the middle chunk ids, just the first and last one.

Reply:
Yes, correct, but if you were to look up the chunk based on its `chunk_id`, you would be able to find the next chunks by selecting chunks with the same `document_id` and a subsequent value of `index`.

Reply:
Ok, so maybe it would make more sense to put the first and last index instead of the id. WDYT?

Reply:
Done.